src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -215,65 +133,21 @@ export function getUBAFeeConfig(
   if (ubaConfig === undefined) {
     throw new Error(`UBA config for blockTag ${blockNumber} not found`);
   }
-  return ubaConfig;
-}
-
-export function getFeesForFlow(
-  flow: UbaFlow,
-  precedingFlowsInBundle: UbaFlow[],
-  bundleState: UBABundleState,
-  chainId: number,
-  tokenSymbol: string
-): {
-  lpFee: BigNumber;
-  relayerBalancingFee: BigNumber;
-  depositBalancingFee: BigNumber;
-  lastRunningBalance: BigNumber;
-  lastIncentiveBalance: BigNumber;
-  netRunningBalanceAdjustment: BigNumber;
-} {
-  const {
-    runningBalance: lastRunningBalance,
-    incentiveBalance: lastIncentiveBalance,
-    netRunningBalanceAdjustment,
-  } = analog.calculateHistoricalRunningBalance(
-    precedingFlowsInBundle,
-    bundleState.openingBalance,
-    bundleState.openingIncentiveBalance,
-    chainId,
-    tokenSymbol,
-    bundleState.config
-  );
-  const { balancingFee: depositBalancingFee } = getDepositFee(
-    flow.amount,
-    lastRunningBalance,
-    lastIncentiveBalance,
-    chainId,
-    bundleState.config
-  );
-  const { balancingFee: relayerBalancingFee } = getRefundFee(
-    flow.amount,
-    lastRunningBalance,
-    lastIncentiveBalance,
-    chainId,
-    bundleState.config
-  );
-  const lpFee = computeLpFeeForRefresh(bundleState.config.getBaselineFee(flow.destinationChainId, flow.originChainId));
 
-  return {
-    lpFee,
-    relayerBalancingFee,
-    depositBalancingFee,
-    lastRunningBalance,
-    lastIncentiveBalance,
-    netRunningBalanceAdjustment,
-  };
+  // Validate omega curves:
+  // - Each curve must have a zero point.
+  const omegaDefaultZeroCurve = ubaConfig.getZeroFeePointOnBalancingFeeCurve(chainId);
+  if (!isDefined(omegaDefaultZeroCurve)) {
+    throw new Error(`Omega curve for chain ${chainId} does not have a zero point`);
+  }
+  return ubaConfig;
 }
-
 /**
  * Returns most recent `maxBundleStates` bundle ranges for a given chain, in chronological ascending order.
  * Will only returns bundle ranges that are subject to UBA rules, which is based on the bundle's start block.
- * @param chainId The chainId to get bundle ranges for.
+ * Additionally, includes potential next bundle block ranges which extend until the latest spoke pool client search
+ * windows, so that the caller can call this function to cover all UBA events.
+ * @param chainId
  * @param maxBundleStates If this is larger than available validated bundles in the HubPoolClient, will throw an error.
  * @param hubPoolBlock Only returns the most recent validated bundles proposed before this block.
```

Comment:
Is there no scenario where we want to look up the block ranges for an older bundle?

If not...`hubPoolBlock` has been removed from the function prototype. 

```suggestion
```

Reply:
I don't see a reason to, this seems like a function we should only be using in the UBA to load UBA bundles. I can even rename this function to start with UBA. WDYT?